src/core/CorePool.ts

```diff
@@ -157,11 +157,36 @@ export class CorePool {
     amount0Requested: JSBI,
     amount1Requested: JSBI
   ): { amount0: JSBI; amount1: JSBI } {
-    // TODO
     this.checkTicks(tickLower, tickUpper);
+
```

Comment:
Please refer to pr #32 memo: Do tick checks as well as position liquidity check in method collect so that we won't set position with 0 liquidity on PositionManager.

Reply:
Because `positionManager.getPositionAndInitIfAbsent` may init position for user mis-input, let's do this check:
```
let potisionForCheck = this.positionManager.getPositionReadonly(positionKey);
assert(JSBI.notEqual(potisionForCheck.tokensOwed0, ZERO) || JSBI.notEqual(potisionForCheck.tokensOwed1, ZERO), "position is not collectable!")
```

Reply:
Ok, make these check is nencessary, nice review